pg_search/src/postgres/build.rs

```diff
@@ -168,24 +168,30 @@ unsafe extern "C" fn build_callback(
     // By running in our own memory context, we can force the memory to be freed with
     // the call to reset().
     unsafe {
-        build_state.memctx.reset();
-        build_state.memctx.switch_to(|_| {
+        build_state.per_row_context.switch_to(|cxt| {
             let mut search_document = writer.schema.new_document();
 
-            row_to_search_document(values, isnull, key_field_name, categorized_fields, &mut search_document).unwrap_or_else(|err| {
-                panic!(
-                    "error creating index entries for index '{}': {err}",
-                    CStr::from_ptr((*(*indexrel).rd_rel).relname.data.as_ptr())
-                        .to_string_lossy()
-                );
-            });
+            row_to_search_document(
+                values,
+                isnull,
+                key_field_name,
+                categorized_fields,
+                &mut search_document,
+            )
+                .unwrap_or_else(|err| {
+                    panic!(
+                        "error creating index entries for index '{}': {err}",
+                        CStr::from_ptr((*(*indexrel).rd_rel).relname.data.as_ptr()).to_string_lossy()
+                    );
+                });
             writer
                 .insert(search_document, item_pointer_to_u64(*ctid))
                 .unwrap_or_else(|err| {
                     panic!("error inserting document during build callback.  See Postgres log for more information: {err:?}")
                 });
+
+            cxt.reset();
```

Comment:
moved the MemoryContext reset to inside the context, for no particular reason other than to be consistent with what InsertState is now doing.